### PR TITLE
Created first version of new component Auction

### PR DESCRIPTION
**Pull request by rayto22**
New component Charity Auction will represent list of lots for purchase by those who want to help with money and get a children's crafts as a keepsake. For now it is just a list without any functionality, except routing to shelter, which made this lot. 
API for this component for now isn't ready, beck-end in process of development. My first version now based on local JSON.
I'll create tests after will be sure, that beck will not change an already created logic of this component.
image

### CODE REVIEW

File: src/app/auction/auction-lot/auction-lot.component.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  align-content: space-between;
+  min-width: 400px;
+}
+
+.lot-info-head-div {
+  display: flex;
+  justify-content: space-between;
+}
+
+.lot-control-div{
```
Comment (ksenya-ta) [inline src/app/auction/auction-lot/auction-lot.component.scss]: add missing space

File: src/app/auction/services/auction.service.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          this.http.get(config.donationItemsApi)
+          )
+        ),
+        map(([listOfLots, children, shelters, dontationItems]: [ActiveLot[], Children[], Shelter[], any]) => {
+          const childrenObj = children.reduce((acc, curr) => ({ [curr.id]: curr, ...acc }), {});
+          const shetlerObj = shelters.reduce((acc, curr) => ({ [curr.id]: curr, ...acc }), {});
+          const dontationItemsObj = dontationItems.reduce((acc, curr) => ({ [curr.id]: curr, ...acc }), {});
+
+          return listOfLots.map((lot: ActiveLot) => {
+            console.log(dontationItemsObj[lot.auctionLotItemID], shetlerObj[childrenObj[lot.orphanID].childrenHouseID]);
```
Comment (ksenya-ta) [inline src/app/auction/services/auction.service.ts]: please remove console.log

File: src/app/auction/services/auction.service.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  public getActiveLots(): Observable<ActiveLot[]> {
+    return this.configService.getConfig().pipe(
+      concatMap((config: Config) =>
+        zip(
+          this.http.get<ActiveLot[]>(config.activeLotsApi),
+          this.http.get<Children[]>(config.childrenApi),
+          this.http.get<Shelter[]>(config.sheltersApi),
+          this.http.get(config.donationItemsApi)
+          )
```
Comment (ksenya-ta) [inline src/app/auction/services/auction.service.ts]: do we really need to make 4 API calls to get the details we need?

Comment (rayto22) [reply src/app/auction/services/auction.service.ts]: I will discuss changing the received data with a backing in the coming days

File: src/app/auction/services/auction.service.ts (hunk 2 above)
Comment (rayto22) [reply src/app/auction/services/auction.service.ts]: Removed.

File: src/app/auction/auction-lot/auction-lot.component.scss (hunk 1 above)
Comment (rayto22) [reply src/app/auction/auction-lot/auction-lot.component.scss]: Added.

**Pull request merged**